src/main/java/org/testng/TestRunner.java

```diff
@@ -1011,8 +1011,9 @@ public void addListener(ITestNGListener listener) {
     }
     if (listener instanceof IExecutionListener) {
       IExecutionListener iel = (IExecutionListener) listener;
-      iel.onExecutionStart();
-      m_configuration.addExecutionListener(iel);
+      if (m_configuration.addExecutionListenerIfAbsent(iel)) {
+        iel.onExecutionStart();
```

Comment:
Out of the scope of the fix but it's not clear for me why are calling the listener here and not at the `run()` time.

Reply:
@juherr - I guess because, when `run()` is executed, we start only from `@BeforeTest` configuration methods and the `@BeforeSuite` have already run. But we need the `IExecutionListener` implementation to be called before anything else. At least that is what I could make out as an explanation, skimming through the codebase.